packages/aws-client/src/DynamoDb.ts

```diff
@@ -137,6 +141,54 @@ export const scan = improveStackTrace(
   }
 );
 
+/**
+ * Do a parallel scan of DynamoDB table using a document client.
+ *
+ * See https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/Scan.html#Scan.ParallelScan.
+ * See [DocumentClient.scan()](https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/DynamoDB/DocumentClient.html#scan-property).
+ *
+ * @param {number} totalSegments
+ *   Total number of segments to divide table into for parallel scanning
+ * @param {DocumentClient.ScanInput} scanParams - Params for the DynamoDB client scan operation
+ * @param {function} processItemsFunc - Function used to process returned items by scan
+ * @returns {Promise}
+ */
+export const parallelScan = async (
+  totalSegments: number,
+  scanParams: DocumentClient.ScanInput,
+  processItemsFunc: (items: DocumentClient.ItemList) => Promise<void>
+) => pMap(
+  range(totalSegments),
+  async (_, segmentIndex) => {
+    let exclusiveStartKey: DocumentClient.Key | undefined;
+
+    const segmentScanParams: DocumentClient.ScanInput = {
+      ...scanParams,
+      TotalSegments: totalSegments,
+      Segment: segmentIndex,
+    };
+
+    /* eslint-disable no-await-in-loop */
+    do {
+      const {
```

Comment:
🤔  Should we consider building in retry behavior here?    I'm assuming from your test runs it didn't wind up being needed, so it's probably YAGNI?

Reply:
You mean a retry of the scan operation itself if it fails? Yeah I never ran into that issue in testing, but not sure if that means we shouldn't add it

Reply:
I'd hate for a migration to outright fail one scan because AWS hiccuped.    That said, if migrations are fast enough now that that's not painful or... expensive perhaps it doesn't matter.

Reply:
5aade542dc619ae178dac71d6e036f4677520541

Reply:
👍🏻